test/integration/server-rendering.spec.tsx

```diff
@@ -8,28 +8,64 @@
 
 /*eslint-disable react/prop-types*/
 
-import React from 'react'
+import React, { FunctionComponent } from 'react'
 import { renderToString } from 'react-dom/server'
 import { createStore } from 'redux'
 import { Provider, connect } from '../../src/index'
+import type { Dispatch, Store } from 'redux'
 
 describe('React', () => {
   describe('server rendering', () => {
-    function greetingReducer(state = { greeting: 'Hello' }, action) {
+    interface ActionType {
+      type: string
+      payload: {
+        greeting: string
+      }
+    }
+    function greetingReducer(
+      state = { greeting: 'Hello' },
+      action: ActionType
+    ) {
       return action && action.payload ? action.payload : state
     }
+    interface GreetingProps {
+      greeting: string
+      greeted: string
+    }
+    const Greeting: FunctionComponent<GreetingProps> = ({
```

Comment:
:hand: I don't think `FunctionComponent` is actually needed here - just use `GreetingProps` after the destructuring. But we can get this in now and tweak stuff later.

(Then again, we could probably replace a lot of our classes with functions anyway)